File: backend/src/main/java/algoview/models/Converter.java
package algoview.models;

import algoview.pojos.DialogDataPojo;
import algoview.pojos.LabelValuePojo;
import lombok.Getter;
import lombok.Setter;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Arrays;

@Getter
@Setter
@Document("converters")
public class Converter {
    @Id
    private String id;
    private String title;
    private String language;
    private String code;

    public DialogDataPojo getDialogData() {
        return new DialogDataPojo("", Arrays.asList(
                new LabelValuePojo("Nazwa", this.title),
                new LabelValuePojo("Kod", this.code, "textarea")
        ));
    }
}

File: backend-node/src/models/Converter.ts
import { ObjectId } from "mongodb";
import { DialogData } from "../structures/DialogData";
import { LabelValue } from "../structures/LabelValue";

export class Converter {

    public dialogData: DialogData

    constructor(
        public title: string = "", 
        public language: string = "", 
        public type: string = "", 
        public code: string = "",

        public id?: ObjectId,
    ) {
        this.title = title;
        this.language = language;
        this.type = type;
        this.code = code;
        this.dialogData = new DialogData(
            "", // title ???
            [
                new LabelValue("Nazwa" , title),
                new LabelValue("Typ zmiennej", type),
                new LabelValue("Język programowania", language),
                new LabelValue("Kod", code, "textarea"),
            ]
        );

        this.id = id ? id : new ObjectId();
    };
}
